app/cdap/components/RecipeDetails/DrawerHeader/index.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ *  Copyright © 2022 Cask Data, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+import { grey } from '@material-ui/core/colors';
+import React from 'react';
+import styled from 'styled-components';
+import CloseIcon from '@material-ui/icons/Close';
+import DeleteOutlineIcon from '@material-ui/icons/DeleteOutline';
+import EditIcon from '@material-ui/icons/Edit';
+import SaveAltIcon from '@material-ui/icons/SaveAlt';
+import { Box, IconButton, Typography } from '@material-ui/core';
+import T from 'i18n-react';
+import { UnderlineIcon } from 'components/RecipeDetails/IconStore/UnderlineSVG';
+
+interface IDrawerHeaderProps {
+  onCloseDetail: () => void;
+}
+
+const MainHeadWrapper = styled(Box)`
+  margin-top: 20px;
+`;
+
+const HeadWrapper = styled(Box)`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const RecipeHeaderLabel = styled(Typography)`
+  font-weight: 400;
+  font-size: 20px;
+  margin-bottom: 6px;
+  color: ${grey[900]};
+  padding: 1px;
+`;
+
+const IconsWrapper = styled(Box)`
+  display: flex;
+  align-items: center;
+`;
+
+const StyledCommonIconButton = styled(IconButton)`
+  padding: 12px;
+`;
+
+const StyledEditIcon = styled(EditIcon)`
+  fill: ${grey[600]};
+  width: 25px;
+  height: 25px;
+`;
+
+const StyledSaveIcon = styled(SaveAltIcon)`
+  fill: ${grey[600]};
+  width: 25px;
+  height: 25px;
+`;
+
+const StyledDeleteIcon = styled(DeleteOutlineIcon)`
+  fill: ${grey[600]};
+  width: 25px;
+  height: 25px;
+`;
+
+const StyledCloseIcon = styled(CloseIcon)`
+  fill: ${grey[600]};
+  width: 25px;
+  height: 25px;
+`;
+
+const VerticalDivider = styled(Box)`
+  width: 1px;
+  height: 25px;
+  border-left: 1px solid ${grey[300]};
+  margin: 0 5px;
+`;
+
+const HeaderWithUnderline = styled(Box)`
+  display: flex;
+  flex-direction: column;
+`;
+
+export default function({ onCloseDetail }: IDrawerHeaderProps) {
+  return (
+    <MainHeadWrapper>
```

Comment:
client has mentioned that we are writing the drawer component and relative components multiple times rather than reusing common components
plz plan on updating this
we have declared header body for drawer in multiple places
at least in this pr if we already have the drawer in multiple places see how we can reuse the same drawer
i don't mind if u update the existing code, but we need a solution to reuse the same component across all prs
**plz update the common drawer component and make it dependable on props to render title, actions and body just by passing props**